utils/deploy/envoy.yaml

```diff
@@ -237,19 +238,21 @@ data:
                                               }
                                             },
                                             {
-                                              "selector": {
-                                                "selector": "source.address"
+                                              "expression": {
+                                                "key": "source.address",
+                                                "value": "source.address"
                                               }
                                             },
                                             {
-                                              "selector": {
-                                                "selector": "request.headers.My-Custom-Header-01"
+                                              "expression": {
+                                                "key": "request.headers['My-Custom-Header-01']",
+                                                "value": "request.headers['My-Custom-Header-01']"
                                               }
                                             },
                                             {
-                                              "selector": {
-                                                "selector": "metadata.filter_metadata.envoy\\.filters\\.http\\.header_to_metadata.user_id",
-                                                "key": "user_id"
+                                              "expression": {
+                                                "key": "user_id",
+                                                "value": "string(getHostProperty(['metadata', 'filter_metadata', 'envoy.filters.http.header_to_metadata', 'user_id']))"
```

Comment:
shouldn't be `auth.somthing`?

Reply:
No this one is coming from the [`header_to_metadata`](https://github.com/Kuadrant/wasm-shim/blob/main/utils/deploy/envoy.yaml#L114-L122) filter proving we can get filter state from non-kuadrant filters